internal/provisioner/objects/dataplane/dataplane.go

```diff
@@ -204,7 +204,7 @@ func desiredContainers(contour *model.Contour, contourImage, envoyImage string)
 					HTTPGet: &corev1.HTTPGetAction{
 						Scheme: corev1.URISchemeHTTP,
 						Path:   "/ready",
-						Port:   intstr.IntOrString{IntVal: int32(8002)},
+						Port:   intstr.IntOrString{IntVal: int32(contour.Spec.RuntimeSettings.Envoy.Health.Port)},
```

Comment:
This reference is nil by default which is resulting in a panic. We'll need to add some nil-checking and defaulting here. Please also add unit tests that cover this for the default and non-default cases.